Remove JsonForms Redux

lucas-koehler opened the pull request:
* Update JsonForms dependencies to ^2.5.0
* Remove redux and react-redux depenencies
* Adapt detail form widget to use plain react JsonForms without redux

Fix #7

sdirix commented on theia-tree-editor/src/browser/detail-form-widget.tsx:
```diff
@@ -36,111 +33,100 @@ const JSON_FORMS_CONTAINER_CSS_CLASS = 'jsonforms-container';
-                            <JsonFormsDispatch />
-                        </JsonFormsReduxContext>
-                    </Provider>
+                    <JsonFormsStyleContext.Provider value={styleContextValue}>
```
Maybe retrieve the context value from a protected method? Then it would be easy to override. Of course it should always return the same object by default.

sdirix commented on theia-tree-editor/src/browser/detail-form-widget.tsx:
```diff
@@ -36,111 +33,100 @@ const JSON_FORMS_CONTAINER_CSS_CLASS = 'jsonforms-container';
 }
+
+// Default vanilla styles extend with theia-specific styling
+const styleContextValue: StyleContext = {
```
export?

sdirix commented on theia-tree-editor/src/browser/detail-form-widget.tsx:
```diff
@@ -36,111 +33,100 @@ const JSON_FORMS_CONTAINER_CSS_CLASS = 'jsonforms-container';
+                            data={data}
+                            schema={schema}
+                            uischema={uiSchema}
+                            cells={vanillaCells}
+                            renderers={vanillaRenderers}
+                            onChange={this.jsonformsOnChange}
+                            config={{
+                                restrict: false,
+                                trim: false,
+                                showUnfocusedDescription: true,
+                                hideRequiredAsterisk: false
+                            }}
+                            refParserOptions={{
+                                dereference: { circular: 'ignore' }
+                            }}
```
`cells`, `renderers`, `config` and `refParserOptions` could be retrieved by a protected method, e.g. `getJsonFormsConfig` which is then added here via `{...getJsonFormsConfig()}`. Then we would be fully flexible. Of course the method should always return the same elements.

sdirix reviewed (approved):
Looks good to me. If you like you can also build in the customization options but we can also tackle this separately.

lucas-koehler commented:
@sdirix Thanks for the review and the customization suggestions. I rebased the changes onto master and added the customization improvements as separate commit https://github.com/eclipse-emfcloud/theia-tree-editor/pull/24/commits/72e9c66aa2f7ca5040b5a8a5ddf00bf7979a75a2

sdirix reviewed (approved):
Works for me! Thanks!
